datalad/support/annexrepo.py

```diff
@@ -335,3 +336,103 @@ def annex_add_to_git(self, files):
             self.cmd_call_wrapper.run(cmd_list, cwd=self.path)
         else:
             self.git_add(files)
+
+    def annex_initremote(self, name, options):
+        """Creates a new special remote
+
+        Parameters:
+        -----------
+        name: str
+            name of the special remote
+        """
+        # TODO: figure out consistent way for passing options + document
+
+        cmd_list = ['git', 'annex', 'initremote', name] + options
+        self.cmd_call_wrapper(cmd_list, cwd=self.path)
+
+    def annex_enableremote(self, name):
+        """Enables use of an existing special remote
+
+        Parameters:
+        -----------
+        name: str
+            name, the special remote was created with
+        """
+
+        cmd_list = ['git', 'annex', 'enableremote', name]
+        self.cmd_call_wrapper(cmd_list, cwd=self.path)
+
+    @normalize_paths
+    def annex_addurl_to_file(self, file, url, options=[]):
+        """Add file from url to the annex.
+
+        Downloads `file` from `url` and add it to the annex.
+        If annex knows `file` already, records that it can be downloaded from `url`.
+
+        Parameters:
+        -----------
+        file: str
```

Comment:
Why? It's easier the way it is. You can use the decorated function with a
str (as desired) and the function itself doesn't have to differentiate
between str and list input, which otherwise every single method would have
to implement in almost the same way. Only problem is correct documentation.

2015-04-15 13:11 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/support/annexrepo.py
> <https://github.com/yarikoptic/datalad/pull/2#discussion_r28410837>:
>
> > +            name, the special remote was created with
> > +        """
> > +
> > +        cmd_list = ['git', 'annex', 'enableremote', name]
> > +        self.cmd_call_wrapper(cmd_list, cwd=self.path)
> > +
> > +    @normalize_paths
> > +    def annex_addurl_to_file(self, file, url, options=[]):
> > +        """Add file from url to the annex.
> > +
> > +        Downloads `file` from `url` and add it to the annex.
> > +        If annex knows `file` already, records that it can be downloaded from `url`.
> > +
> > +        Parameters:
> > +        -----------
> > +        file: str
>
> May be just make that normalize smarter... pass string if was a string
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yarikoptic/datalad/pull/2/files#r28410837>.
>